src/libraries/System.Text.Json/gen/Reflection/TypeExtensions.cs

```diff
@@ -26,33 +31,54 @@ public static string GetCompilableName(this Type type)
             {
                 StringBuilder sb = new();
 
-                string fullName = type.FullName;
-                int backTickIndex = fullName.IndexOf('`');
-
-                string baseName = fullName.Substring(0, backTickIndex);
+                if (!string.IsNullOrEmpty(type.Namespace))
+                {
+                    sb.Append(type.Namespace);
+                    sb.Append('.');
+                }
 
-                sb.Append(baseName);
+                int argumentIndex = 0;
+                AppendTypeChain(sb, type, type.GetGenericArguments(), ref argumentIndex);
 
-                sb.Append('<');
+                compilableName = sb.ToString();
+            }
 
-                Type[] genericArgs = type.GetGenericArguments();
-                int genericArgCount = genericArgs.Length;
-                List<string> genericArgNames = new(genericArgCount);
+            compilableName = compilableName.Replace("+", ".");
```

Comment:
is string replace the only way to detect this? is there any way to not have to do this?

Reply:
This is old code for non-generic types: https://github.com/dotnet/runtime/pull/71619/files/25e4f6793c7bd216163069b1dc0834795b131380#diff-6c3f829d936e6e251e4900b01bc9a99dfb2dd30782bfd905758f591871333db4L54

Yes, it is possible to avoid it but it will be necessary to modify the following section
https://github.com/dotnet/runtime/pull/71619/files/25e4f6793c7bd216163069b1dc0834795b131380#diff-6c3f829d936e6e251e4900b01bc9a99dfb2dd30782bfd905758f591871333db4R28

Reply:
fair enough, if it's simple feel free to do it but treat this as optional feedback since we've already done that before